descartes_samplers/include/descartes_samplers/evaluators/impl/gantry_euclidean_distance_edge_evaluator.hpp

```diff
@@ -25,63 +25,25 @@ DESCARTES_IGNORE_WARNINGS_POP
 
 #include <descartes_samplers/evaluators/gantry_euclidean_distance_edge_evaluator.h>
 
-namespace
-{
-template <typename FloatType>
-static void considerEdge(const FloatType* start,
-                         const FloatType* end,
-                         std::size_t dof,
-                         std::size_t next_idx,
-                         typename descartes_light::LadderGraph<FloatType>::EdgeList& out)
-{
-  FloatType cost = 0.0;
-  for (std::size_t i = 2; i < static_cast<std::size_t>(dof); ++i)
-  {
-    cost += std::pow(end[i] - start[i], FloatType(2));
-  }
-
-  out.emplace_back(cost, next_idx);
-}
-
-}  // namespace
-
 namespace descartes_light
 {
 template <typename FloatType>
 GantryEuclideanDistanceEdgeEvaluator<FloatType>::GantryEuclideanDistanceEdgeEvaluator(int dof)
-  : dof_(static_cast<std::size_t>(dof))
+  : EdgeEvaluator<FloatType>(static_cast<std::size_t>(dof))
 {
 }
 
 template <typename FloatType>
-bool GantryEuclideanDistanceEdgeEvaluator<FloatType>::evaluate(
-    const Rung_<FloatType>& from,
-    const Rung_<FloatType>& to,
-    std::vector<typename LadderGraph<FloatType>::EdgeList>& edges)
+std::pair<bool, FloatType> GantryEuclideanDistanceEdgeEvaluator<FloatType>::considerEdge(const Rung_<FloatType>&,
+                                                                                         const FloatType* start,
+                                                                                         const Rung_<FloatType>&,
+                                                                                         const FloatType* end)
 {
-  const auto n_start = from.data.size() / dof_;
-  const auto n_end = to.data.size() / dof_;
-
-  // Allocate
-  edges.resize(n_start);
-
-  for (std::size_t i = 0; i < n_start; ++i)
-  {
-    const auto* start_vertex = from.data.data() + dof_ * i;
-    for (std::size_t j = 0; j < n_end; ++j)
-    {
-      const auto* end_vertex = to.data.data() + dof_ * j;
-
-      // Consider the edge:
-      considerEdge(start_vertex, end_vertex, dof_, j, edges[i]);
-    }
-  }
-
-  for (const auto& rung : edges)
-    if (!rung.empty())
-      return true;
+  FloatType cost = 0.0;
+  for (std::size_t i = 2; i < this->dof_; ++i)
```

Comment:
There are a lot of assumptions being made here:
- The gantry has two external axes
- The gantry axes are the first two values in the list
- The joint distance of the gantry axes is not considered

We need to add these assumptions to the documentation at the very least. Ultimately, we should probably parameterize them with the arguments to the constructor.

Why is the motion of the first two axes of the gantry irrelevant here?

Reply:
Based on experience it works better to minimize robot motion and let the gantry do most of the work if it can. Though this is old and not used anymore. I good with marking it deprecated or removing it completely.

Reply:
We don't necessarily need to delete it (unless you really think it needs to go); we just need to document these assumptions on the class definition so we can easily remember how it works

Reply:
I think it has little value since most will likely use there own because MoveIt and Tesseract have there own way of defining kinematics. Tesseract implements its own which allow any number of gantry joints and I would assume if other motion planning libraries would do the same.

Reply:
I update your Issue #5 to include documentation items that need to be addressed.